Objects/longobject.c

```diff
@@ -4399,8 +4427,45 @@ PyLong_GetInfo(void)
 int
 _PyLong_Init(void)
 {
+    PyInterpreterState *interp = PyThreadState_GET()->interp;
     /* initialize long_info */
     if (Long_InfoType.tp_name == 0)
         PyStructSequence_InitType(&Long_InfoType, &long_info_desc);
+    interp->long_max_str_digits = Py_LongMaxStrDigits;
+    if (interp->long_max_str_digits == -1) {
+        interp->long_max_str_digits = _PY_LONG_DEFAULT_MAX_STR_DIGITS;
+    }
     return 1;
 }
+
+
+void
+_PyLongMaxStrDigits_Init(void)
+{
+    char *env;
+    const char *endptr;
+    long maxdigits;
+
+
+    if (Py_LongMaxStrDigits >= 0 ||
+        !Py_LongMaxStrDigitsFlag)
+        return;
+
+    env = Py_GETENV("PYTHONINTMAXSTRDIGITS");
+    if (env && *env != '\0') {
+        errno = 0;
+        maxdigits = strtol(env, (char **)&endptr, 10);
+        if (*endptr != '\0' || errno == ERANGE || maxdigits < INT_MIN || maxdigits > INT_MAX ||
+            !((maxdigits == 0) || (maxdigits >= _PY_LONG_MAX_STR_DIGITS_THRESHOLD))) {
+#define STRINGIFY(VAL) _STRINGIFY(VAL)
+#define _STRINGIFY(VAL) #VAL
+            Py_FatalError(
+                "PYTHONINTMAXSTRDIGITS: invalid limit; must be >= "
+                STRINGIFY(_PY_LONG_MAX_STR_DIGITS_THRESHOLD)
+                " or 0 for unlimited.");
+#undef _STRINGIFY
+#undef STRINGIFY
+        }
+        Py_LongMaxStrDigits = (int)maxdigits;
```

Comment:
This is the other place I'd check that `maxdigits` is 0 or positive.

Reply:
I don't understand, `maxdigits` is checked for 0 or positive on line 4459. I suppose it is theoretically possible that `_PY_LONG_MAX_STR_DIGITS_THRESHOLD` is changed to a negative, but that would probably lead to multiple issues.
Note that [`Py_FatalError`](https://docs.python.org/2.7/c-api/sys.html?highlight=fatalerror#c.Py_FatalError) doesn't return, it will kill the process.

Reply:
I may have been misreading this conditional. To confirm then, `maxdigits` has be either `0` or a value greater than `_PY_LONG_MAX_STR_DIGITS_THRESHOLD`, or the process will abort?

That should work then.